hamt/hamt.go

```diff
@@ -408,6 +415,35 @@ func (ds *Shard) ForEachLink(ctx context.Context, f func(*ipld.Link) error) erro
 	})
 }
 
+func (ds *Shard) makeAsyncTrieGetLinks(cb func(*ipld.Link) error) dag.GetLinks {
```

Comment:
Could we add some documentation for this method? Especially explaining what is the callback argument `cb` (also please consider expanding the name to something more descriptive) supposed/expected to do.